packages/python/plotly/plotly/__init__.py

```diff
@@ -99,9 +112,27 @@ def plot(data_frame, kind, **kwargs):
     if kind == "box":
         new_kwargs = {k: kwargs[k] for k in kwargs if k not in ["by"]}
         return box(data_frame, **new_kwargs)
-    if kind in "hist":
+    if kind in ["hist", "histogram"]:
         new_kwargs = {k: kwargs[k] for k in kwargs if k not in ["by", "bins"]}
         return histogram(data_frame, **new_kwargs)
+    if kind == "violin":
+        return violin(data_frame, **kwargs)
+    if kind == "strip":
+        return strip(data_frame, **kwargs)
+    if kind == "funnel":
+        return funnel(data_frame, **kwargs)
+    if kind == "density_contour":
```

Comment:
I don't really understand how `df.plot(kind='density_contour')` is supposed to work. Does it suppose it's a wide-mode or long-mode dataframe?

Reply:
it supports wide mode, yes. it works just like scatter basically, which works just like line :)

Reply:
ok